metrics/test_driver/flutter_web_driver.dart

```diff
@@ -1,56 +1,142 @@
+import 'dart:async';
 import 'dart:io';
 
-import 'common/arguments/driver_tests_parser.dart';
-import 'common/process_manager/process_manager.dart';
-import 'util/file_utils.dart';
-
-Future<void> main(List<String> arguments) async {
-  final args = DriverTestsArgs(arguments);
-
-  final Directory workingDir = Directory(args.workDir);
-  final Directory logsDir = Directory(args.logsDir);
-
-  if (!workingDir.existsSync()) workingDir.createSync();
-  if (!logsDir.existsSync()) logsDir.createSync();
-
-  final workingDirPath = workingDir.path;
-
-  final processManager = ProcessManager(
-    args.port,
-    logsDir.path,
-    verbose: args.verbose,
-    quiet: args.quiet,
-  );
-
-  final seleniumFileName = await FileUtils.downloadSelenium(workingDirPath);
-  await FileUtils.downloadChromeDriver(workingDirPath);
-  await FileUtils.downloadFirefoxDriver(workingDirPath);
-
-  print('Running $seleniumFileName...');
-
-  await processManager.startSelenium(seleniumFileName, workingDirPath);
-
-  print("Selenium server is up, running flutter application...");
-
-  await processManager.startFlutterApp().catchError((error) {
-    exitCode = 1;
-    processManager.exitApp();
-  });
-
-  stdout.done.asStream().listen((_) => processManager.exitApp());
-
-  print("Application is up, running tests...");
-
-  await processManager
-      .startDriverTests(browserName: args.browserName)
-      .catchError((_) {
-    processManager.exitApp();
-  });
-
-  print(
-      "Flutter logs are stored in ${logsDir.absolute.uri}${ProcessManager.flutterLogsFileName}.log file");
-  print(
-      "Driver logs are stored in ${logsDir.absolute.uri}${ProcessManager.flutterLogsFileName}.log file");
-
-  processManager.exitApp();
+import 'arguments/model/driver_test_arguments.dart';
+import 'arguments/parser/driver_test_arguments_parser.dart';
+import 'cli/flutter/runner/flutter_drive_process_runner.dart';
+import 'cli/flutter/runner/flutter_run_process_runner.dart';
+import 'cli/selenium/runner/selenium_process_runner.dart';
+import 'cli/selenium/selenium.dart';
+import 'common/config/logs_file_config.dart';
+import 'common/logger/logger.dart';
+import 'process_manager/process_manager.dart';
+
+/// Runs the application and driver tests for this application.
+class FlutterWebDriver {
+  final DriverTestArguments _args;
+  final ProcessManager _processManager = ProcessManager();
+
+  /// Creates the [FlutterWebDriver].
+  ///
+  /// [args] is the application arguments to run with.
+  /// See [DriverTestArgumentsParser] to get all available arguments.
```

Comment:
```suggestion
  /// See [DriverTestArgumentsParser] for all supported arguments.
```
? or we can just skip this?